modules/ui/src/main/scala/lucuma/ui/optics/TruncatedBigDecimal.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.optics
+
+import cats.Eq
+import lucuma.core.optics.SplitEpi
+
+/**
+ * A wrapper around a BigDecimal that is limited to a specified
+ * number of decimals places.
+ *
+ * The Dec type parameter must be a Singleton Int, which is enforced
+ *    by the compiler. The compiler cannot, however, keep you from
```

Comment:
Could this help?
https://github.com/fthomas/singleton-ops#supported-relational-operations

Reply:
Oh! That's cool. I'll use that.